### PR TITLE
haxe: 4.3.4 -> 4.3.5

### PR DESCRIPTION
**Pull request by bwkam**
## Description of changes

update haxe
https://haxe.org/download/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JeremiahSecrist** [comment]
Result of `nixpkgs-review pr 332277` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>haxe</li>
    <li>haxePackages.format</li>
    <li>haxePackages.heaps</li>
    <li>haxePackages.hlopenal</li>
    <li>haxePackages.hlsdl</li>
    <li>haxePackages.hxcpp</li>
    <li>haxePackages.hxcs</li>
    <li>haxePackages.hxjava</li>
    <li>haxePackages.hxnodejs_4</li>
  </ul>
</details>

**Comment by JeremiahSecrist** [comment]
it looks the issue persists upon rerunning it. I recommend testing locally as well with the command I used above.

Result of `nixpkgs-review pr 332277` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>haxe</li>
    <li>haxePackages.format</li>
    <li>haxePackages.heaps</li>
    <li>haxePackages.hlopenal</li>
    <li>haxePackages.hlsdl</li>
    <li>haxePackages.hxcpp</li>
    <li>haxePackages.hxcs</li>
    <li>haxePackages.hxjava</li>
    <li>haxePackages.hxnodejs_4</li>
  </ul>
</details>

**Comment by bwkam** [comment]
> it looks the issue persists upon rerunning it. I recommend testing locally as well with the command I used above.
> 
> Result of `nixpkgs-review pr 332277` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 9 packages failed to build:

I tried running the haxe binary locally before pushing, and it worked alright. I haven't tried nixpkgs-review though.

**Comment by bwkam** [comment]
oops, seems like I accidentally fetched the hash for nightly haxe, not 4.3.5.

**Comment by JeremiahSecrist** [comment]
we have builds!
Result of `nixpkgs-review pr 332277` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>haxe</li>
    <li>haxePackages.format</li>
    <li>haxePackages.heaps</li>
    <li>haxePackages.hlopenal</li>
    <li>haxePackages.hlsdl</li>
    <li>haxePackages.hxcpp</li>
    <li>haxePackages.hxcs</li>
    <li>haxePackages.hxjava</li>
    <li>haxePackages.hxnodejs_4</li>
  </ul>
</details>

File: pkgs/development/compilers/haxe/default.nix
Comment (JeremiahSecrist) [inline pkgs/development/compilers/haxe/default.nix:1]: Oh! Can we move this to `pkgs/by-name/ha`? They changed the repo structure a while back and are gradually migrating everything. I'm not sure if this needs to be a separate PR/Commit, so let's wait for someone else to weigh in.

Comment (bwkam) [reply pkgs/development/compilers/haxe/default.nix:1]: if it's really important, I can move it. But I prefer the latter.

Comment (drupol) [reply pkgs/development/compilers/haxe/default.nix:1]: No it's not.

Comment (JeremiahSecrist) [reply pkgs/development/compilers/haxe/default.nix:1]: then LGTM

**Review by JeremiahSecrist — APPROVED** [review]
